src/connector.rs

```diff
@@ -0,0 +1,630 @@
+            amount,
+            hex::encode(address)
+        ));
+        self.ft.internal_withdraw_eth(address, amount);
+    }
+
+    /// Withdraw from NEAR accounts
+    pub fn withdraw_near(&mut self) {
+        sdk::assert_one_yocto();
+        #[cfg(feature = "log")]
+        sdk::log("Start withdraw NEAR");
+        let args =
+            WithdrawCallArgs::try_from_slice(&sdk::read_input()[..]).expect(ERR_FAILED_PARSE);
+        let recipient_address = validate_eth_address(args.recipient_id);
+        let res = WithdrawResult {
+            recipient_id: recipient_address,
+            amount: args.amount,
+            eth_custodian_address: self.contract.eth_custodian_address,
+        }
+        .try_to_vec()
+        .unwrap();
+        // Burn tokens to recipient
+        let predecessor_account_id = String::from_utf8(sdk::predecessor_account_id()).unwrap();
+        self.burn_near(predecessor_account_id, args.amount);
+        // Save new contract data
+        self.save_contract();
+        sdk::return_output(&res[..]);
+    }
+
+    /// Return total supply of NEAR + ETH
+    pub fn ft_total_supply(&self) {
+        let total_supply = self.ft.ft_total_supply();
+        sdk::return_output(&total_supply.to_string().as_bytes());
+        #[cfg(feature = "log")]
+        sdk::log(&format!("Total supply: {}", total_supply));
+    }
+
+    /// Return total supply of NEAR
+    pub fn ft_total_supply_near(&self) {
+        let total_supply = self.ft.ft_total_supply_near();
+        sdk::return_output(&total_supply.to_string().as_bytes());
+        #[cfg(feature = "log")]
+        sdk::log(&format!("Total supply NEAR: {}", total_supply));
+    }
+
+    /// Return total supply of ETH
+    pub fn ft_total_supply_eth(&self) {
+        let total_supply = self.ft.ft_total_supply_eth();
+        sdk::return_output(&total_supply.to_string().as_bytes());
+        #[cfg(feature = "log")]
+        sdk::log(&format!("Total supply ETH: {}", total_supply));
+    }
+
+    /// Return balance of NEAR
+    pub fn ft_balance_of(&self) {
+        let args =
+            BalanceOfCallArgs::try_from_slice(&sdk::read_input()[..]).expect(ERR_FAILED_PARSE);
+        let balance = self.ft.ft_balance_of(&args.account_id);
+        sdk::return_output(&balance.to_string().as_bytes());
+        #[cfg(feature = "log")]
+        sdk::log(&format!(
+            "Balance of NEAR [{}]: {}",
+            args.account_id, balance
+        ));
+    }
+
+    /// Return balance of ETH
+    pub fn ft_balance_of_eth(&self) {
+        let args =
+            BalanceOfEthCallArgs::try_from_slice(&sdk::read_input()[..]).expect(ERR_FAILED_PARSE);
+        let balance = self.ft.internal_unwrap_balance_of_eth(args.address);
+        #[cfg(feature = "log")]
+        sdk::log(&format!(
+            "Balance of ETH [{}]: {}",
+            hex::encode(args.address),
+            balance
+        ));
+        sdk::return_output(&balance.to_string().as_bytes());
+    }
```

Comment:
As I mentioned earlier, this method duplicates the already existing [`get_balance`](https://github.com/aurora-is-near/aurora-engine/blob/95c80ff003f607f162c0b750f4551eb3a4b29aa4/src/lib.rs#L282) method. Probably we need to get rid of one of those.
@artob what do you think?